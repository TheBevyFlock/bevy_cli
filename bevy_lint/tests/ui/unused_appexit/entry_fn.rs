#![feature(register_tool)]
#![register_tool(bevy)]
//~v NOTE: the lint level is defined here
#![deny(bevy::unused_appexit)]

use bevy::prelude::*;

//~v HELP: set the return type of `fn main()`
fn main() {
    App::new().run();
    //~^ ERROR: called `App::run()` without handling the returned `AppExit`
    //~| NOTE: `App::run()` returns `AppExit`, which is used to determine whether the app exited successfully or not
    //~| HELP: `AppExit` implements `Termination`, so it can be returned directly from `fn main()`
    //~| HELP: return the result of `App::run()`
}
