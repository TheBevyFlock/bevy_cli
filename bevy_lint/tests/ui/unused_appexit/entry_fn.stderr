error: called `App::run()` without handling the returned `AppExit`
  --> tests/ui/unused_appexit/entry_fn.rs:10:16
   |
10 |     App::new().run();
   |                ^^^^^
   |
   = note: `App::run()` returns `AppExit`, which is used to determine whether the app exited successfully or not
   = help: `AppExit` implements `Termination`, so it can be returned directly from `fn main()`
note: the lint level is defined here
  --> tests/ui/unused_appexit/entry_fn.rs:4:9
   |
4  | #![deny(bevy::unused_appexit)]
   |         ^^^^^^^^^^^^^^^^^^^^
help: return the result of `App::run()`
   |
10 |     return App::new().run();
   |     ~~~~~~~~~~~~~~~~~~~~~~~
help: set the return type of `fn main()`
   |
9  | fn main() -> AppExit {
   |           ++++++++++

error: aborting due to 1 previous error

